mirrord/agent/src/steal/http/reversible_stream.rs

```diff
@@ -19,25 +22,79 @@ use super::error::HttpTrafficError;
 pub(crate) struct ReversibleStream<const HEADER_SIZE: usize> {
     #[pin]
     stream: TcpStream,
+
     header: [u8; HEADER_SIZE],
+
+    /// We could be writing less then `HEADER_SIZE` bytes into `header`, in two cases:
+    /// 1. Got a timeout before reading as `HEADER_SIZE` bytes.
+    /// 2. Read an EOF after less than `HEADER_SIZE` bytes.
+    /// So we need to always now how many bytes we actually have.
+    header_len: usize,
+
+    /// How many bytes out of the `header` were already read by the owner of self.
```

Comment:
Not really clear

Reply:
What about now https://github.com/metalbear-co/mirrord/pull/936/commits/103974dae475f1be2e4e40a2f419a2a1458c8c8e?

Reply:
Great thanks